cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/dataset/SearchResults.java

```diff
@@ -16,26 +16,35 @@
 
 package io.cdap.cdap.data2.metadata.dataset;
 
+import java.util.LinkedList;
 import java.util.List;
 
 /**
  * Represents a list of {@link MetadataEntry} that match a search query in the metadata index, along with a
  * list of cursors to start subsequent searches from.
  */
 public class SearchResults {
-  private final List<MetadataEntry> results;
+  private final List<MetadataResultEntry> results;
   private final List<String> cursors;
 
 
-  SearchResults(List<MetadataEntry> results, List<String> cursors) {
+  SearchResults(List<MetadataResultEntry> results, List<String> cursors) {
     this.results = results;
     this.cursors = cursors;
   }
 
-  public List<MetadataEntry> getResults() {
+  public List<MetadataResultEntry> getResults() {
     return results;
   }
 
+  public List<MetadataEntry> getEntries() {
+    List<MetadataEntry> entries = new LinkedList<>();
+    for (MetadataResultEntry m : this.results) {
```

Comment:
Can be done with one liner:

`return results.stream().map(MetadataResultEntry::getMetadataEntry).collect(Collectors.toList());`

Reply:
implemented in new commit